docs - add topics for btree_gin and pg_trgm modules

lisakowen opened the pull request:
docs for: https://github.com/greenplum-db/gpdb/pull/12928, https://github.com/greenplum-db/gpdb/pull/12927, https://github.com/greenplum-db/gpdb/pull/12929 (6X_STABLE)

will be backported to 6X_STABLE.

GPORCA support of these new modules is not clear to me...

Sasasu commented on gpdb-doc/dita/ref_guide/modules/btree_gin.xml:
```diff
@@ -0,0 +1,45 @@
+      <body>
+        <p>Because Greenplum Database provides no hash operator class for the 
+          <codeph>money</codeph> data type, using the <codeph>btree_gin</codeph> module
+          on this type is a no-op.</p>
```
hash operator used by `distributed by`, not btree_gin.
any `create table FOO(X money)` in Greenplum will meet that problem, not only `btree_gin`.

I think we can remove this part.

Sasasu reviewed (approved):
on 7X GPORCA can optimize queries that access index with op_class without error, means ORCA supports this module.
for 6X GPORCA will fall back to planner.

lisakowen commented:
thank you for reviewing, @Sasasu .  i removed the limitations section for 7/master.  i added a commit to add the limitations section back in with the gporca limitation for 6x.
